tests/linux_benchmarks/iperf_benchmark_test.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2020 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for netperf_benchmark."""
+
+import json
+import os
+import unittest
+from absl import flags
+import mock
+import itertools
+
+from perfkitbenchmarker import benchmark_spec
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.linux_benchmarks import iperf_benchmark
+
+FLAGS = flags.FLAGS
+
+class IperfBenchmarkTestCase(unittest.TestCase):
+
+  def testIperfParseResults(self):
+
+    vm_spec = mock.MagicMock(spec=benchmark_spec.BenchmarkSpec)
+    vm0 = mock.MagicMock()
+    vm1 = mock.MagicMock()
+    vm_spec.vms = [vm0, vm1]
+    path = os.path.join(os.path.dirname(__file__), '..', 'data', 'iperf_results.txt')
+    outfile = open(path, 'r')
+    iperf_stdout_list = outfile.read().split('#')
+
+    for vm in vm_spec.vms:
+      vm.RemoteCommand.side_effect = [(i, '') for i in iperf_stdout_list]
+      vm.machine_type = 'mock_machine_1'
+      vm.zone = 'antarctica-1a'
+
+    results1 = iperf_benchmark._RunIperf(vm0, vm1, '10.128.0.2', 1, 'INTERNAL', 'UDP')
```

Comment:
Please make each test case it's own method.

You can probably include the output in the method as a triple quoted string rather than having a standalone file.